src/tests/java/picard/sam/markduplicates/MarkDuplicatesTagRepresentativeeadIndexTester.java

```
@@ -0,0 +1,124 @@
+
+                final String key = samRecordToDuplicatesFlagsKey(record);
+                if (!this.duplicateFlags.containsKey(key)) {
+                    System.err.println("DOES NOT CONTAIN KEY: " + key);
+                }
+                Assert.assertTrue(this.duplicateFlags.containsKey(key));
+                final boolean value = this.duplicateFlags.get(key);
+                this.duplicateFlags.remove(key);
+                if (value != record.getDuplicateReadFlag()) {
+                    System.err.println("Mismatching read:");
+                    System.err.print(record.getSAMString());
+                }
+                Assert.assertEquals(record.getDuplicateReadFlag(), value);
+                if (testRepresentativeReads) {
+                    if (expectedRepresentativeIndexMap.containsKey(indexInFile) && expectedSetSizeMap.containsKey(record.getReadName())){
+                        Assert.assertEquals(record.getAttribute("DI"), expectedRepresentativeIndexMap.get(indexInFile));
+                        Assert.assertEquals(record.getAttribute("DS"), expectedSetSizeMap.get(record.getReadName()));
+                    }
+                }
+                indexInFile+=1;
+            }
+            CloserUtil.close(reader);
+
+            // Ensure the program output the same number of records as were read in
+            Assert.assertEquals(outputRecords, this.getNumberOfRecords(), ("saw " + outputRecords + " output records, vs. " + this.getNumberOfRecords() + " input records"));
+
+            // Check the values written to metrics.txt against our input expectations
+            final MetricsFile<DuplicationMetrics, Comparable<?>> metricsOutput = new MetricsFile<DuplicationMetrics, Comparable<?>>();
+            try{
+                metricsOutput.read(new FileReader(metricsFile));
+            }
+            catch (final FileNotFoundException ex) {
+                System.err.println("Metrics file not found: " + ex);
```

Should you maybe throw this exception after catching it?